Sources/Private/MainThread/LayerContainers/CompLayers/TextCompositionLayer.swift

```diff
@@ -92,8 +98,18 @@ final class TextCompositionLayer: CompositionLayer {
   let textDocument: KeyframeInterpolator<TextDocument>?
 
   let textLayer = CoreTextRenderLayer()
-  var textProvider: AnimationTextProvider
+  var textProvider: AnimationKeypathTextProvider
   var fontProvider: AnimationFontProvider
+  weak var rootAnimationLayer: MainThreadAnimationLayer?
+
+  lazy var fullAnimationKeypath: AnimationKeypath = {
+    // Individual layers don't know their full keypaths, so we have to delegate
+    // to the `MainThreadAnimationLayer` to search the layer hierarchy and find
+    // the full keypath (which includes this layer's parent layers)
+    rootAnimationLayer?.keypath(for: self)
```

Comment:
This was tricky -- I suppose this is why the Main Thread rendering engine didn't previously work like this